skiko/src/jvmMain/kotlin/org/jetbrains/skiko/SkiaLayer.jvm.kt

```diff
@@ -95,10 +95,21 @@ actual open class SkiaLayer internal constructor(
     }
 
     override fun removeNotify() {
-        dispose()
+        disposeSkiaResources()
```

Comment:
Also we should call full `dispose` (with `backedLayer.dispose()`), as it frees resources on macOs. Without it, we will have a memory leak, if user won't manually call `dispose`.

Reply:
In the case of the "recreation window" we must not remove the `backedLayer`, otherwise we lost all the added event listeners and, unfortunately, the case of the "recreation window" was not intended to remove any other native resources other than the window itself (AWT/Swing just re-adds all the components in place). So in this case, there should be no memory leak (by the way on macos I have not encountered this "recreation window" case).

And, of course, there is a problem with swing interop - when we want to remove `ComposePanel`, we have to remove `backedLayer` manually.

Reply:
> In the case of the "recreation window" we must not remove the backedLayer, otherwise we lost all the added event listeners

We don't recreate class `HardwareLayer`, we just call `nativeInit`. In that case we don't lose listeners.

> as not intended to remove any other native resources other than the window itself

`addNotify`/`removeNotify` exists for the case if third-party developers want to init/free their resources without providing additional `dispose` method to the user.

Imagine we don't have `dispose` method (it can be our plan), similar to ComposePanel.

>  So in this case, there should be no memory leak

But it is, I see it with my eyes :). User don't call `dispose` - we don't free Window on macOs. I am still talking about the usual case, not the `recreation case`.

Reply:
> But it is, I see it with my eyes :)
How?:) As I say - there is no recreation window case on macos - only if you manually call `remove` `SkiaLayer` you get memory leak and I say:
> when we want to remove `ComposePanel`, we have to remove (dispose) `backedLayer` manually.
But I get what you mean - I'm implementing it right now.

Reply:
> How?:) 
```
val frame = JFrame()
val layer = SkiaLayer()
frame.contentPane.add(layer)
frame.isVisible = true
frame.dispose()
```

> we have to remove (dispose) backedLayer manually.

That is exactly what I am talking about. If we call `dispose` in `removeNotify`, we don't have to remove it manually. And that was our intention, when we overrode `removeNotify`, to not call `SkiaLayer.dispose` manually.

Reply:
By the way, we also can deprecate `dispose` method, an leave it empty, if it will simplify the PR.

Reply:
> By the way, we also can deprecate dispose method,

For now it should be discussed because of now we have `expect detach()` which in JVM is `dispose()`

Reply:
Check new commit.